go/adbc/driver/snowflake/driver_test.go

```diff
@@ -325,31 +325,26 @@ type SnowflakeTests struct {
 	stmt   adbc.Statement
 }
 
-func (suite *SnowflakeTests) SetupSuite() {
+func (suite *SnowflakeTests) SetupTest() {
```

Comment:
why the switch from `SetupSuite` (once for all the tests) to `SetupTest` (once for each test)?

Reply:
This allows all tests to use the SetupDriver and TeardownDriver methods defined on the DriverQuirks. This came to my attention because only the tests in `validation.go` which were using Setup/TeardownDriver caught the memory leak. This allows us to have the same test setup for the driver tests as well. Without this, the driver was getting created with one checked allocator and each test would create another one, preventing a full view of memory allocations.